File: python/courtlistener/api/agreements.py
"""
Agreements API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class AgreementsAPI(BaseAPI):
    """API for accessing agreements data from financial disclosures."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "agreements/"
    
    def list(
        self,
        financial_disclosure: Optional[int] = None,
        judge: Optional[int] = None,
        **kwargs
    ) -> Dict[str, Any]:
        """
        List agreement records.
        
        Args:
            financial_disclosure: Filter by financial disclosure ID
            judge: Filter by judge ID
            **kwargs: Additional query parameters
        
        Returns:
            API response with paginated list of agreements
        """
        params = {}
        if financial_disclosure is not None:
            params['financial_disclosure'] = financial_disclosure
        if judge is not None:
            params['judge'] = judge
        
        params.update(kwargs)
        return self.client.get(self.endpoint, params=params)
    
    def get(self, agreement_id: int) -> Dict[str, Any]:
        """
        Get a specific agreement by ID.
        
        Args:
            agreement_id: Agreement ID
        
        Returns:
            Agreement data
        """
        return self.client.get(f"{self.endpoint}{agreement_id}/")
    
    def paginate(
        self,
        financial_disclosure: Optional[int] = None,
        judge: Optional[int] = None,
        **kwargs
    ):
        """
        Get paginated agreements.
        
        Args:
            financial_disclosure: Filter by financial disclosure ID
            judge: Filter by judge ID
            **kwargs: Additional query parameters
        
        Returns:
            PageIterator for agreements
        """
        params = {}
        if financial_disclosure is not None:
            params['financial_disclosure'] = financial_disclosure
        if judge is not None:
            params['judge'] = judge
        
        params.update(kwargs)
        return self.client.paginate(self.endpoint, params=params)

File: php/src/CourtListener/Api/Agreements.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * Agreements API endpoint
 */
class Agreements extends BaseApi
{
    protected string $endpoint = 'agreements/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List agreements with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listAgreements(array $params = []) {
        return $this->list($params);
    }

    /**
     * Get a specific agreements by ID
     *
     * @param int|string $id Agreements ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getAgreements($id, array $params = []) {
        return $this->get($id, $params);
    }

    /**
     * Search agreements
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchAgreements(array $params = []) {
        return $this->search($params);
    }
}
